### ISSUE TITLE
Bug: exp map on the sphere does not handle tensor input

### ISSUE THREAD

**Issue by tibopp** [geomstats/geomstats#2051]
### Describe the bug

I want to use the exp map with tensor data, but the following example yields an error, which I assume is due to bad handling of tensor data.

What am I missing ?

### Steps/Code to Reproduce

sphere = hs.Hypersphere(dim=2)
x0 = torch.from_numpy(sphere.random_uniform(5))
print(sphere.metric.exp(torch.tensor([0.1, 0.1, 0.1]), x0))

### Expected Behaviour

Should return a tensor of same dimension as x

### Actual Behaviour

An error that is avoided if x0 is converted to numpy array, but I'd like to keep it at a tensor

### Your environment

```shell
Windows 11, python 3.12.19, geomstats 2.8.0
```

**Comment by luisfpereira** [comment]
Hi @tibopp. If you start your script by:

```python
import os
os.environ["GEOMSTATS_BACKEND"] = "pytorch"
```

and then do:

```python
sphere = hs.Hypersphere(dim=2)
x0 = sphere.random_uniform(5)
print(sphere.metric.exp(torch.tensor([0.1, 0.1, 0.1]), x0))
```

it will work. The reason is that we do all the operations with the selected "GEOMSTATS_BACKEND", so all the output will be consistent (moreover, the functions expect the correct array type).

Please reopen if the fix does not work.

**Comment by tibopp** [comment]
Thank you, this works fine !
